service/http/response.go

```diff
@@ -59,3 +61,44 @@ func (r *Response) Write(w http.ResponseWriter) error {
 
 	return nil
 }
+
+func handlePushHeaders(h map[string][]string) ([]string, map[string][]string) {
+	var p []string
+	pushHeader, ok := h["http2-push"]
+	if !ok {
+		return p, h
+	}
+
+	for _, v := range pushHeader {
+		p = append(p, v)
+	}
+
+	delete(h, "http2-push")
+
+	return p, h
+}
+
+func handleTrailers(h map[string][]string) map[string][]string {
+	trailers, ok := h["trailer"]
+	if !ok {
+		return h
+	}
+
+	zp := regexp.MustCompile(` *, *`)
```

Comment:
I decided to use regular expressions to split string into components. Alternatively `strings.Split()` can be used and parts can be trimmed independently later. Not sure what would be better and open to suggestions.

Reply:
I would prefer to use Split here. It will also help to avoid regexp compilation on every trailed response.

Reply:
Thanks. Will address this shortly.